packages/test/test-service-load/src/FileLogger.ts

```diff
@@ -3,80 +3,85 @@
  * Licensed under the MIT License.
  */
 
-import crypto from "crypto";
 import fs from "fs";
 
 import { ITelemetryBufferedLogger } from "@fluid-internal/test-driver-definitions";
 import { ITelemetryBaseEvent, LogLevel } from "@fluidframework/core-interfaces";
-import { assert, LazyPromise } from "@fluidframework/core-utils/internal";
+import { assert } from "@fluidframework/core-utils/internal";
 import { createChildLogger } from "@fluidframework/telemetry-utils/internal";
 
 import { pkgName, pkgVersion } from "./packageVersion.js";
 
-export class FileLogger implements ITelemetryBufferedLogger {
-	private static readonly loggerP = new LazyPromise<FileLogger>(async () => {
-		if (process.env.FLUID_TEST_LOGGER_PKG_SPECIFIER !== undefined) {
+// This test expects that if a certain env variable is specified, it points to a package that will
+// pollute the global with a "getTestLogger" when imported.  getTestLogger is actually expected to be
+// "instantiateTestLogger" in practice (it creates a new one, rather than retrieving an existing one).
+// Generally speaking, this global logger type will be the one that actually knows how to log to a real
+// destination (i.e. aria-logger).
+// TODO: Consider injecting a logger rather than relying on an environment variable and dynamic import.
+// TODO: Consider just exporting the function and importing it directly rather than polluting the global.
+const maybeInstantiateGlobalLoggerType = async () => {
+	if (process.env.FLUID_TEST_LOGGER_PKG_SPECIFIER !== undefined) {
+		if (getTestLogger === undefined) {
```

Comment:
I was a bit confused by the refactor but finally got what it's doing. I'd suggest an explicit comment noting that checking `getTestLogger` seemingly before the `await import` that would define it is to optimize repeated calls to `maybeInstantiateGlobalLoggerType` (even if we don't technically do that).

Reply:
Added a comment to that effect in latest.  Realistically it should be benign to double import here as it should just reassign the same function to getTestLogger, but it felt silly :)